wraps/account-abstraction/docs/readme.md

```diff
@@ -0,0 +1,127 @@
+      connections: new Connections({
+        networks: {
+          goerli: new Connection({
+            provider: "RPC_URL", // Add your RPC URL here
+            signer: new Wallet("PRIVATE_KEY"), // Add your private key here
+          }),
+        },
+        defaultNetwork: "goerli",
+      }),
+    }),
+    // This adds the datetime plugin
+    "wrap://ens/datetime.polywrap.eth": dateTimePlugin({}) as IWrapPackage,
+  })
+  // This URL should be replaced with the URL of the AA Wrap
+  .addEnv("wrap://wrapper/account-abstraction", {
+    connection: {
+      networkNameOrChainId: "goerli", // Determines which network you're using
+    },
+  });
+
+// Build the config
+const config = builder.build();
+
+// You can now use the Polywrap Client to invoke the AA Wrap
+const client = new PolywrapClient(config);
+
+const invocationResult = client.invoke({
+  uri: "wrap://wrapper/account-abstraction",
+  args: { ... }
+})
+```
+
+## Examples
+
+### Get Safe Address
+
+Since your Safe address is deterministic based on your wallet's address and a nonce, you can determine it ahead-of-time:
+
+```javascript
+const addressResult = await client.invoke({
+  uri: "wrap://ens/aa.safe.wraps.eth:core@0.1.0",
+  method: "getSafeAddress",
+  args: {
+    config: {
+      saltNonce: "0x258802387238728372837283771" // Replace this with your own salt nonce
+    }
+  }
+});
+```
+
+### Relay Sponsored Transaction
+
+Let's assume we have a smart contract with a `store(uint256 num)` function deployed to the address `0x56535D1162011E54aa2F6B003d02Db171c17e41e`.
+If we want to, for example, invoke this function, we first need to encode it using our [Ethers Utils Wrap](https://github.com/polywrap/ethers):
+
+```javascript
+const encodeResult = await client.invoke({
+  uri: "wrap://ens/ethers.wraps.eth:utils@0.1.0",
+  method: "encodeFunction",
+  args: {
+    method: "function store(uint256 num) public",
+    args: [
+      "14"
+    ]
+  }
+});
+```
+
+The Account Abstraction Wrapper's `relayTransaction` method allows you to make a gasless transaction by performing the following steps:
+1. It checks whether there's a deployed Safe contract on your predicted address.
+2. It deploys a Safe contract to the predicted address if there isn't one.
+3. It executes the gasless (sponsored) transaction using the Gelato Relay.
+
+```javascript
+await client.invoke({
+  uri: "wrap://ens/aa.safe.wraps.eth:core@0.1.0",
+  method: "relayTransaction",
+  args: {
+    transaction: {
+      to: "0x56535D1162011E54aa2F6B003d02Db171c17e41e", // Address of the contract you're invoking
+      value: "0",
+      data: encodeResult.data, // Invocation data.
+      operation: "0"
+    },
+    options: {
+      gasLimit: "300000",
+      isSponsored: true
+    },
+    config: {
+      // Replace this with your own salt nonce. This is used to deploy your safe contract if there is none.
+      saltNonce: "0x258802387238728372837283771"
```

Comment:
nit: salt nonce does not need to be defined, if it is not sent, the wrapper will take `Date.now()` as salt (and as hex)

Reply:
Added to comments